pkg/network/master/master.go

```diff
@@ -166,33 +172,27 @@ func (master *OsdnMaster) checkClusterNetworkAgainstClusterObjects() error {
 func clusterNetworkChanged(obj *networkapi.ClusterNetwork, old *networkapi.ClusterNetwork) (bool, error) {
 	changed := false
 
-	if old.Network != obj.Network {
+	if len(old.ClusterNetworks) != len(obj.ClusterNetworks) {
 		changed = true
+	}
 
-		_, newNet, err := net.ParseCIDR(obj.Network)
-		if err != nil {
-			return true, err
-		}
-		newSize, _ := newNet.Mask.Size()
-		oldBase, oldNet, err := net.ParseCIDR(old.Network)
-		if err != nil {
-			// Shouldn't happen, but if the existing value is invalid, then any change should be an improvement...
-		} else {
-			oldSize, _ := oldNet.Mask.Size()
-
-			// oldSize and newSize are, eg the "16" in "10.1.0.0/16", so
-			// "newSize < oldSize" means the new network is larger
-			if !(newSize < oldSize && newNet.Contains(oldBase)) {
-				return true, fmt.Errorf("cannot change clusterNetworkCIDR to a value that does not include the existing network.")
+	for _, oldCIDR := range old.ClusterNetworks {
+		found := false
+		for _, newCIDR := range obj.ClusterNetworks {
+			if newCIDR.CIDR == oldCIDR.CIDR && newCIDR.HostSubnetLength == oldCIDR.HostSubnetLength {
+				found = true
 			}
 		}
+		if found == false {
+			changed = true
+			break
+		}
 	}
-	if old.HostSubnetLength != obj.HostSubnetLength {
-		return true, fmt.Errorf("cannot change the hostSubnetLength of an already-deployed cluster")
-	}
+
 	if old.ServiceNetwork != obj.ServiceNetwork {
 		return true, fmt.Errorf("cannot change the serviceNetworkCIDR of an already-deployed cluster")
 	}
+
 	if old.PluginName != obj.PluginName {
 		changed = true
 	}
```

Comment:
We can quickly return in some cases without requiring comparing old and new cluster networks.

```
if old.ServiceNetwork != obj.ServiceNetwork {
  return true, fmt.Errorf(...)
} else if old.PluginName != obj.PluginName {
  return true, nil
} else if len(old.ClusterNetworks) != len(obj.ClusterNetworks) {
  return true, nil
} else {
  // Compare old and new cluster networks and set changed
  return changed, nil
}
return false, nil
```